MREUnityRuntime/MREUnityRuntimeLib/Core/ActorManager.cs

```diff
@@ -103,12 +104,21 @@ internal bool HasActor(Guid? id)
 
         internal void ProcessActorCommand(Guid actorId, NetworkCommandPayload payload, Action onCompleteCallback)
         {
-            if (!_actorCommandQueues.TryGetValue(actorId, out ActorCommandQueue queue))
+            _actorCommandQueues.GetOrCreate(actorId, () =>
             {
-                queue = new ActorCommandQueue(actorId, _app);
+                var queue = new ActorCommandQueue(actorId, _app);
                 _actorCommandQueues.Add(actorId, queue);
```

Comment:
>_actorCommandQueues.Add(actorId, queue); [](start = 16, length = 40)

Do you need to call Add here, or does GetOrCreate handle the addition?

Reply:
GetOrCreate does everything for you, it's a util extension function that I copied over from Altspace ages ago.

---
In reply to: [304130289](https://github.com/microsoft/mixed-reality-extension-unity/pull/88#discussion_r304130289) [](ancestors = 304130289)